src/projections.jl

```diff
@@ -211,7 +212,7 @@ function projection_gradient_on_set(::DefaultDistance, v::AbstractVector{T}, ::M
     λ, U = LinearAlgebra.eigen(X)
 
     # if all the eigenvalues are >= 0
-    if all(>=(0), λ)
+    if all(λ .≥ zero(T))
```

Comment:
```suggestion
    if all(λi ≥ zero(λi) for λi in λ)
```

Reply:
a generator will boil down to a for loop and avoid the allocation of a broadcast